src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioWorkspaceImpl.cs

```diff
@@ -81,8 +81,8 @@ protected void InitializeStandardVisualStudioWorkspace(SVsServiceProvider servic
             var projectTracker = new VisualStudioProjectTracker(serviceProvider, this.Services);
 
             // Ensure the document tracking service is initialized on the UI thread
-            var documentTrackingService = this.Services.GetService<IDocumentTrackingService>();
-            var documentProvider = new RoslynDocumentProvider(projectTracker, serviceProvider, documentTrackingService);
+            var documentTrackingService = this.Services.GetService<IDocumentTrackingService>() as VisualStudioDocumentTrackingService;
```

Comment:
At this layer, we would expect this cast to always pass, right?  Can we make it a hard case instead of an `as`, otherwise we might later find some strange bug where the service was null and not understand why.